src/hooks/useMapSetup.ts

```diff
@@ -0,0 +1,273 @@
+import { watch } from '@arcgis/core/core/reactiveUtils';
+import MapView from '@arcgis/core/views/MapView';
+import WebMap from '@arcgis/core/WebMap';
+import BasemapToggle from '@arcgis/core/widgets/BasemapToggle.js';
+import Home from '@arcgis/core/widgets/Home';
+import Track from '@arcgis/core/widgets/Track';
+import { useFirebaseAnalytics } from '@ugrc/utah-design-system';
+import debounce from 'lodash.debounce';
+import { useEffect, useRef } from 'react';
+import { getCoarseLocation, INITIAL_MAP_ZOOM } from '../components/mapUtils';
+import type { LayerNames } from '../context/FirebaseRemoteConfigsProvider';
+import { getUrlParameter, setUrlParameter } from '../utilities/UrlParameters';
+import { useFilter } from './useFilter';
+import useRemoteConfigs from './useRemoteConfigs';
+
+type LayerNameKey = keyof LayerNames;
+
+export function useMapSetup(
+  mapNodeRef: React.RefObject<HTMLDivElement | null>,
+  trayIsOpen: boolean,
+  useMyLocationOnLoad: boolean,
+  isSmallScreen: boolean,
+  onFeatureIdentify: (graphic: __esri.Graphic | null) => void,
+  popupRoot: HTMLDivElement,
+  feedbackButtonRef: React.RefObject<HTMLDivElement | null>,
+  zoomButtonRef: React.RefObject<HTMLDivElement | null>,
+  setCenter: (center: number[]) => void,
+) {
+  const getConfig = useRemoteConfigs();
+  const map = useRef<WebMap | null>(null);
+  const mapView = useRef<MapView>(null);
+  const layers = useRef<Record<LayerNameKey, __esri.FeatureLayer | null>>({
+    routeTypes: null,
+    trafficStress: null,
+    trafficSignals: null,
+    otherLinks: null,
+  });
+  const mapIsInitialized = useRef<boolean>(false);
+  const highlightHandle = useRef<__esri.Handle>(null);
+  const { dispatch } = useFilter();
+
+  const logEvent = useFirebaseAnalytics();
+
+  // set up map
+  useEffect(() => {
+    if (!mapNodeRef.current || mapIsInitialized.current || !getConfig) {
+      return;
+    }
+
+    const initializeMap = async () => {
+      console.log('setting up the map');
```

Comment:
```suggestion
```

Reply:
I actually like this to make sure that I don't start initializing the map multiple times.